### PR TITLE
[feaLib] Support name table

### PR DESCRIPTION
**Pull request by khaledhosny**
Needs some tests, but I though to get feedback first since this is my first try with feaLib code.

### CODE REVIEW

File: Lib/fontTools/feaLib/parser.py [hunk 1]
```diff
@@ -712,6 +714,77 @@ def parse_table_head_(self, table):
...
+                              platEncID, langID, string)
+
+    def unescape_mac_name_string(self, string):
+        def unescape(match):
+            n = match.group(0)[1:]
+            # Can't figure a way that works for both Python 2 and 3
+            if bytes == str:
+                c = chr(int(n, 16)).decode('mac_roman')
+            else:
+                c = bytes([int(n, 16)]).decode('mac_roman')
```
Comment (adrientetar) [inline Lib/fontTools/feaLib/parser.py]: I think you want to use bytechr() in fontTools.misc.py23.

Comment (khaledhosny) [inline Lib/fontTools/feaLib/parser.py]: Thanks, I looked into py23 but I obviously missed this.

**Comment by khaledhosny** [comment]
I’ve added a couple of test files and squashed all the commits together.

File: Lib/fontTools/feaLib/parser.py [hunk 2]
```diff
@@ -712,6 +714,77 @@ def parse_table_head_(self, table):
...
+                                      self.cur_token_location_)
+
+    def parse_nameid_(self):
+        assert self.cur_token_ == "nameid", self.cur_token_
+        location, nameID = self.cur_token_location_, self.expect_number_()
+        if nameID > 255:
+            raise FeatureLibError("Expected name id < 255",
+                                  self.cur_token_location_)
+        if 1 <= nameID <= 6:
+            # TODO: issue a warning
```
Comment (anthrotype) [inline Lib/fontTools/feaLib/parser.py]: I suggest you import `logging` and create a `log` instance at the top of the parser module:

```python
import logging
log = logging.getLogger(__name__)
```

and then here you call `log.warning(...)` to send an appropriate warning message.

**Comment by anthrotype** [comment]
LGTM! Thanks Khaled!

/cc @brawer

**Comment by brawer** [comment]
:+1:

**Pull request merged**